### PR TITLE
Prettier: switch to single-quotes by default

### PR DESCRIPTION
**Pull request by chalin**
My original inclination (supported by @svrnm and @cartermp) was to opt for single quotes (https://github.com/open-telemetry/opentelemetry.io/pull/2290#discussion_r1098882987), but I reverted after seeing that YAML code blocks were being changed. Well, in retrospect (and given the situation we're facing in https://github.com/open-telemetry/opentelemetry.io/pull/2372#pullrequestreview-1306451171), I'd rather opt for single quotes for all (including JS code blocks, given that it's part of the OTel-JS standard).

WDYT @svrnm @cartermp?

This PR:

- Contributes to #1063
- Enables Prettier's single-quotes option
- Drops quotes from YAML code blocks in those situations where quotes are unnecessary

Two situations where I wasn't sure if the YAML key value type was important, so I didn't drop the quotes (cc @TylerHelmuth):

- https://github.com/open-telemetry/opentelemetry.io/commit/208befc5131ab827bd1c0da41dcca5e063c9bba4#r101481607
- https://github.com/open-telemetry/opentelemetry.io/commit/01c2c7fb0ef7089bf3ea511c7eb0c8bd52d9333b#r101482142

**Edit**: I added comments to this PR matching the two listed above. Sorry for the duplication!

Preview: https://deploy-preview-2377--opentelemetry.netlify.app

### CODE REVIEW

File: content/en/docs/collector/scaling.md [hunk 1]
```diff
@@ -171,7 +171,7 @@ kind: Pod
 metadata:
   name: my-microservice
   annotations:
-    sidecar.opentelemetry.io/inject: "true"
+    sidecar.opentelemetry.io/inject: 'true'
```
Comment (chalin) [inline content/en/docs/collector/scaling.md:174]: @TylerHelmuth - can we drop the quotes here?

File: content/en/docs/k8s-operator/automatic.md [hunk 2]
```diff
@@ -164,7 +164,7 @@ spec:
     - baggage
   sampler:
     type: parentbased_traceidratio
-    argument: "1"
+    argument: '1'
```
Comment (chalin) [inline content/en/docs/k8s-operator/automatic.md:167]: @TylerHelmuth - can we drop the quotes here (and at line 237)?

**Review by cartermp — APPROVED** [review]
I don't personally see issues with this as-is, but would like confirmation of the single quote messes with the otel yaml or not

File: content/en/docs/collector/scaling.md (hunk 1 above)
Comment (TylerHelmuth) [reply content/en/docs/collector/scaling.md:174]: No, but single quotes work.

File: content/en/docs/k8s-operator/automatic.md (hunk 2 above)
Comment (TylerHelmuth) [reply content/en/docs/k8s-operator/automatic.md:167]: I'm not sure if they can be dropped, but single quotes will work.

**Review by reyang — APPROVED** [review]

**Pull request merged**